### ISSUE TITLE
Vagrantfile - how to use it?

### ISSUE THREAD

**Issue by consideRatio** [jupyterhub/zero-to-jupyterhub-k8s#821]
@manics I tried to use the Vagrantfile to setup a VM with minikube etc within it, but I got stuck with the image pulling part. None of the images available on my computer were available to the VM created by the Vagrantfile. So in order for it to function, I had to specify a file that was pushed to the web.

I tried to learn how to connect my localhost's docker to the VM-host docker in a way that allowed the VM-host to find images on my localhost and pull from them. I failed :'(

**Comment by manics** [comment]
I originally added the Vagrantfile because travis is a pain when it comes to figuring out deployment setup. I haven't tried it recently, but I'd expect running the equivalent of the commands in .travis.yml would work, I'll give it a go and let you know.

**Comment by manics** [comment]
Short version: The original idea was
```
cd ci
vagrant up
vagrant ssh
```
and run the build/test commands. It looks there have been a few changes to the CI setup since I created it. I can update it and add some better instructions if you think it's useful? Or do you think it's better to remove it completely, especially if you're thinking of moving to a different test platform?

**Comment by consideRatio** [comment]
Let us leave it, I'll add some documentation about it!

I learned a lot about it today. I almost got things to work on the trusty distribution, but not really. Apparently `trusty` is coming with Python 3.4, and cant be upgraded easily to Python 3.5+. But dev-dependencies fail to install with Python 3.4. So... I changed the VM to ubuntu bionic coming with Python 3.6 instead and got a lot of things to work.

I got it working now I think, I just need to learn about PYTHONPATH the env variable and where pip3 will install packages etc. I think...
